src/crimson/osd/heartbeat.h

```diff
@@ -39,7 +45,8 @@ class Heartbeat : public crimson::net::Dispatcher {
   seastar::future<> remove_peer(osd_id_t peer);
 
   seastar::future<> send_heartbeats();
-  seastar::future<> send_failures();
+  seastar::future<> send_failures(failure_queue_t failure_queue);
```

Comment:
could you move `send_heartbeats()`, `send_failures()` and `heartbeat_check()` down to the `private` section? as they are not part of the public interface of this class. and might want to keep `clock` and `failure_queue_t` close to where they are used.

Reply:
could you reconsider `heartbeat_check()`?